### ISSUE TITLE
Scene-composer: Abnormal Number of Requests for History Retrieval for Custom Component Type

### ISSUE THREAD

**Issue by S4lem** [awslabs/iot-app-kit#2918]
**Describe the bug**
I am using the scene-composer package to visualize a twin. Additionally, I am utilizing a custom component type (dataReader). The connection to the dataReader is functioning well. However, despite setting a viewport of 5000 ms, I observe that it sends hundreds of requests per minut. You can see the number of requests sent in a few seconds in the screenshot below.

Below is the config.js file (inspired by the example react app)
[https://github.com/awslabs/iot-app-kit/blob/main/examples/react-app/src/configs.ts](https://github.com/awslabs/iot-app-kit/blob/main/examples/react-app/src/configs.ts) :

```
import { getEnvCredentials } from './getEnvCredentials';

export const awsCredentials = getEnvCredentials();

// The pre-configured values below is based on the sample workspace created from
// https://github.com/aws-samples/aws-iot-twinmaker-samples, update them for your own workspace.
export const region = 'eu-west-1';
export const workspaceId = process.env.REACT_APP_TWINMAKER_WORKSPACE_NAME;

// SceneViewer
export const sceneId = "XXXX-scene"; 


export const entityQueries = [
  {
    entityId: "XXXX", 
    componentName: process.env.REACT_APP_TWINMAKER_COMPONENT_NAME,
    properties: [
      { propertyName: 'Property1'}, 
      { propertyName: 'Property2'}, 
      { propertyName: 'Property3'}, 
     ....
    ], 
  },

];
export const dataBindingTemplate = {
  sel_entity: (entityQueries[0]).entityId,
};


```
And here is the code that calls the SceneViewer Component:

```
import { useCallback, useEffect } from 'react';

import { SceneViewer as SceneViewerComp } from '@assistxr/iak-scene-composer-custom';
import { dataSource } from '../../../dataSource';
import { sceneId, entityQueries, dataBindingTemplate } from '../../../twinConfig';
import { Box } from '@chakra-ui/react';

const sceneLoader = dataSource.s3SceneLoader(sceneId);
const sceneMetadataModule = dataSource.sceneMetadataModule(sceneId);

const queries = [
    ...entityQueries.map((q) => dataSource.query.timeSeriesData(q)),
];

const SceneViewer = () => {
    const onSelectionChanged = useCallback((e) => {
        console.log('onSelectionChanged event fired with data: ', e);
        ...
    }, []);

    const onWidgetClick = useCallback((e) => {
        console.log('onWidgetClick event fired with data: ', e);
        ...

    }, []);

    ....
    return (
        <Box
            display={"flex"}
            alignItems={"center"}
            justifyContent={"center"}
            position={"relative"}
            width={"full"}
            height={"full"}
        >
            <div className="SceneViewer">
                <SceneViewerComp
                    style={{
                        borderRadius: "15px"
                    }}
                    sceneLoader={sceneLoader}
                    config={{
                        dracoDecoder: {
                            enable: true,
                            path: 'https://www.gstatic.com/draco/versioned/decoders/1.5.3/', // path to the draco files
                        },
                        colorTheme: "light",
                    }}
                    onSelectionChanged={onSelectionChanged}
                    sceneMetadataModule={sceneMetadataModule}
                    onWidgetClick={onWidgetClick}
                    queries={queries}
                    dataBindingTemplate={dataBindingTemplate}
                    sceneComposerId={sceneId}
                    viewport={{
                        duration: "5s"
                    }}
                    activeCamera="Top-Right View"
                />
            </div>
        </Box>
    );
};

export default SceneViewer;
```

I have tested with the SiteWise connector and do not encounter this issue; the requests are sent every 5000 ms as expected.

**Expected behavior**

I expect the requests to be sent every 5000 ms (5 seconds), matching the viewport setting. Instead, hundreds of requests are being sent per second.

**Screenshots**

You can observe the number of requests sent in a few seconds in the screenshot from the DevTools below.

![Capture d'écran 2024-07-19 121309](https://github.com/user-attachments/assets/e8944fb7-12bc-4807-94c5-0822a0d5c628)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Microsoft Edge
 - Version 126.0.2592.81 

scene-composer package version : 10.8.0

**Comment by chejimmy** [comment]
Thank you for bringing this to our attention.I am checking with the team regarding this issue 🙏

**Comment by haweston** [comment]
You should be seeing a GetPropertyValue request for each entity in your query list every refresh interval.  So more entities tracked more requests. This should be true for any twinmaker component types not just custom ones.

But the refresh interval isn't set by the viewport duration.  Viewport duration is used to set a timespan of how back into the past (starting from the present) to check for date in.  So a viewport duration of 5 minutes will only use data from the last 5 minutes for results.

To control the frequency of data polling instead use dataBindingQueryRefreshRate in the SceneViewer config you referenced above like:
```
config={{
          dracoDecoder: {
            enable: true,
            path: 'https://www.gstatic.com/draco/versioned/decoders/1.5.3/', // path to the draco files
          },
          dataBindingQueryRefreshRate: 10000,
        }}
```

to set it to repull all registered queries every 10 seconds.

Let me know if that leaves any open questions to follow up on.

**Comment by S4lem** [comment]
After some research, I found the issue.
The custom component type used to return `"nextToken": ""` by default. This triggered an infinite ping-pong game between iot-app-kit and the custom connector.
Something I do not understand is why Twinmaker reconstructs the nextToken even if my field is an empty string.

I had to completely remove it from the output to fix the issue.

Thank you for your answer and your time.
